tools/src/_utils/AjvErrorsParser.ts

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+import { Ajv2019 as AJV, ErrorsTextOptions, ErrorObject } from 'ajv/dist/2019'
+import _ from 'lodash'
+
+export default class AjvErrorsParser {
+  private readonly ajv: AJV
+  private readonly options: ErrorsTextOptions
+
+  constructor(ajv: AJV, options: ErrorsTextOptions = {}) {
+    this.ajv = ajv
+    this.options = { separator: ' --- ', ...options }
+  }
+
+  parse(errors: ErrorObject[] | undefined | null): string {
+    const error_groups = this.#group_errors(errors ?? [])
+    const parsed_errors = [
+      this.#prohibited_property_error(error_groups.prohibited),
+      this.#required_property_error(error_groups.required),
+      this.#enum_error(error_groups.enum),
+      ...error_groups.others
+    ].filter(e => e != null) as ErrorObject[]
```

Comment:
I think `_.compact()` will do the same, possibly cleaner.

Reply:
I wish it was like `Array.compact()` because this is more `()` nesting 😅